Bottom: f72632775425a478f279480f6c137486c3ee3ae3
Top:    996e355f7dacbde458af2d8a3a12a3784a1ddc21
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-24 14:42:24 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index e6353c2..296b875 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -306,6 +306,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
